native/KotlinIOS/KotlinIOS/ViewController.swift

```diff
@@ -6,10 +6,20 @@ class ViewController: UIViewController {
     @IBOutlet var arrivalStationScrollWheel: UIPickerView!
 
     @IBOutlet var fareSearchButton: UIButton!
+    @IBOutlet var searchStatusBar: UILabel!
     @IBOutlet var trainsTable: UITableView!
+    
 
     @IBAction func trainSearchButtonPressed(_ sender: Any) {
-        presenter.onSearchClicked(initialStation: "KGX", ultimateStation: "EDB", timeUTCString: nil)
+        searchStatusBar.text = "searching ..."
+        validTrains = []
+        trainsTable.reloadData()
+        
+        let initialStationIndex = departureStationScrollWheel.selectedRow(inComponent: 0)
+        let finalStationIndex = arrivalStationScrollWheel.selectedRow(inComponent: 0)
+        presenter.onSearchClicked(initialStation: stationList[initialStationIndex],
+                                  ultimateStation: stationList[finalStationIndex],
+                                  timeUTCString: nil)
```

Comment:
I'd prefer these to be called departureStation and arrivalStation everywhere, rather than calling them initial/ultimate in a couple of places